### PR TITLE
feat: referred users query

### PR DESCRIPTION
**Pull request by sshanzel**
Fetch all the users that have been referred by the login user.

From the entity itself, the `createdAt` is not nullable, hence, updated the types.

Made the `createdAt` to be always required as it is part of the query for fetching the list which does not necessarily need to be displayed in FE.

### CODE REVIEW

**Review by rebelchris — APPROVED** [review]
Looks good to me

**Comment by idoshamun** [comment]
I couldn't find the DR, but do we need the users themselves or just the count?

**Comment by rebelchris** [comment]
@idoshamun We need [list](https://dailydotdev.atlassian.net/wiki/spaces/HAN/pages/659161109/Referral+program+and+profile+UX+improvements#Query-to-fetch-referred-users) of users.
Screenshot 2023-11-09 at 13 44 49

**Review by idoshamun — APPROVED** [review]

File: __tests__/users.ts [hunk 1]
```diff
@@ -310,6 +310,53 @@ describe('query userStats', () => {
...
+          image
+        }
+      }
+    }
+  }`;
+
+  it('should not allow unauthenticated users', () =>
+    testQueryErrorCode(client, { query: QUERY }, 'UNAUTHENTICATED'));
+
+  it('should return users that have been referred by the logged in user', async () => {
```
Comment (idoshamun) [inline __tests__/users.ts:331]: Seems like we can merge this and the test below

**Comment by idoshamun** [comment]
@rebelchris thanks!

**Pull request merged**